programs/account-compression/src/instructions/append_leaves.rs

```diff
@@ -65,34 +71,71 @@ pub fn process_append_leaves_to_merkle_trees<'a, 'b, 'c: 'info, 'info>(
             .push(&leaves[i]);
     }
 
-    let mut changelog_events = Vec::new();
+    #[cfg(target_os = "solana")]
+    light_heap::GLOBAL_ALLOCATOR.log_total_heap("append_leaves: merkle tree map");
+
+    let mut changelog_events = Vec::with_capacity(leaves.len());
     for (mt, leaves) in merkle_tree_map.values() {
         let merkle_tree_account = AccountLoader::<StateMerkleTreeAccount>::try_from(mt).unwrap();
         let merkle_tree_pubkey = merkle_tree_account.key();
         let mut merkle_tree = merkle_tree_account.load_mut()?;
 
         check_registered_or_signer::<AppendLeaves, StateMerkleTreeAccount>(&ctx, &merkle_tree)?;
 
-        msg!("inserting leaves: {:?}", leaves);
+        // msg!("inserting leaves: {:?}", leaves);
```

Comment:
probably makes sense to remove along side all logs that we have onchain or feature get with test-sbf as well